ChaosMod/Effects/db/Peds/PedsSeriousKamikazes.cpp

```diff
@@ -0,0 +1,144 @@
+{
+	Vector3 randomCoord;
+	randomCoord.z = startCoord.z;
+	float groundZ;
+	for (int i = 0; i < 10; i++) {
+		if (g_Random.GetRandomInt(0, 1) % 2 == 0)
+		{
+			randomCoord.x = startCoord.x + g_Random.GetRandomInt(minOffset, maxOffset);
+		}
+		else
+		{
+			randomCoord.x = startCoord.x - g_Random.GetRandomInt(minOffset, maxOffset);
+		}
+		if (g_Random.GetRandomInt(0, 1) % 2 == 0)
+		{
+			randomCoord.y = startCoord.y + g_Random.GetRandomInt(minOffset, maxOffset);
+		}
+		else
+		{
+			randomCoord.y = startCoord.y - g_Random.GetRandomInt(minOffset, maxOffset);
+		}
+		randomCoord.z = startCoord.z;
+		if (GET_GROUND_Z_FOR_3D_COORD(randomCoord.x, randomCoord.y, randomCoord.z, &groundZ, false, false))
+		{
+			randomCoord.z = groundZ;
+			break;
+		}
+	}
+	return randomCoord;
+}
+
+static void OnStop()
+{
+	for (Ped ped : kamikazeEnemies)
+	{
+		SET_PED_AS_NO_LONGER_NEEDED(&ped);
+	}
+	kamikazeEnemies.clear();
+
+	for (Object bomb : bombObject)
+	{
+		SET_OBJECT_AS_NO_LONGER_NEEDED(&bomb);
+	}
+	bombObject.clear();
+}
+
+static void OnStart()
+{
+	static const Hash playerGroup = GET_HASH_KEY("PLAYER");
+	ADD_RELATIONSHIP_GROUP("_HOSTILE_SERIOUS_KAMIKAZES", &relationshipGroup);
+	SET_RELATIONSHIP_BETWEEN_GROUPS(5, relationshipGroup, playerGroup);
+	SET_RELATIONSHIP_BETWEEN_GROUPS(5, playerGroup, relationshipGroup);
+	SET_RELATIONSHIP_BETWEEN_GROUPS(0, relationshipGroup, relationshipGroup);
+}
+
+static void OnTick()
+{
+	Ped playerPed = PLAYER_PED_ID();
+	Vector3 playerPos = GET_ENTITY_COORDS(playerPed, false);
+	int current_time = GET_GAME_TIMER();
+
+	for (std::list<Entity>::iterator it = kamikazeEnemies.begin(); it != kamikazeEnemies.end(); )
+	{
+		Ped kamikaze = *it;
+		Vector3 kamikazePos = GET_ENTITY_COORDS(kamikaze, false);
+		if (IS_PED_DEAD_OR_DYING(kamikaze, false) || IS_PED_INJURED(kamikaze) || GET_DISTANCE_BETWEEN_COORDS(playerPos.x, playerPos.y, playerPos.z, kamikazePos.x, kamikazePos.y, kamikazePos.z, false) > 100.f)
+		{
+			SET_ENTITY_HEALTH(kamikaze, 0, 0);
+			WAIT(300);
+			SET_ENTITY_ALPHA(kamikaze, 0, true);
+			SET_PED_AS_NO_LONGER_NEEDED(&kamikaze);
+			DELETE_PED(&kamikaze);
+			it = kamikazeEnemies.erase(it);
+			WAIT(0);
+		}
+		else
+		{
+			it++;
+		}
+	}
+
+	if (kamikazeEnemies.size() < maxKamikazesToSpawn && current_time > spawnTimer + 2000)
+	{
+		spawnTimer = current_time;
+		Vector3 spawnPos = getRandomOffsetCoord(playerPos, 10, 25);
+		USE_PARTICLE_FX_ASSET("core");
+		START_PARTICLE_FX_NON_LOOPED_AT_COORD("exp_air_molotov", spawnPos.x, spawnPos.y, spawnPos.z, 0, 0, 0, 2, true, true, true);
+		WAIT(300);
+		Hash kamikazeHash = GET_HASH_KEY("a_m_y_musclbeac_01");
+		Hash bombHash = GET_HASH_KEY("imp_prop_bomb_ball");
+		LoadModel(kamikazeHash);
+		Ped ped = CREATE_PED(-1, kamikazeHash, spawnPos.x, spawnPos.y, spawnPos.z, 0, true, false);
+		SET_BLOCKING_OF_NON_TEMPORARY_EVENTS(ped, true);
+		SET_PED_RELATIONSHIP_GROUP_HASH(ped, relationshipGroup);
+		SET_PED_HEARING_RANGE(ped, 9999.f);
```

Comment:
Maybe even make them put their hands up like in serious Sam 😅

Reply:
I couldn't find a way to have hands up while running. "TASK_HANDS_UP" function freezes peds. And playing an animation isn't successful so far. I'm still looking for it though

Reply:
static const char* animDict = "missminuteman_1ig_2";
	REQUEST_ANIM_DICT(animDict);
	while (!HAS_ANIM_DICT_LOADED(animDict))
	{
		WAIT(0);
	}
	TASK_PLAY_ANIM(ped, animDict, "handsup_enter", 8, 8, -1, 50, 0, false, false, false);

Reply:
I tried that already, the problem I have is that they wont have their hands up while running. If I remove TASK_FOLLOW_TO_OFFSET_OF_ENTITY, then it works. Even after the effect has ended the kamikazes walk away with their hands up.
But as soon as I'm using the TASK_FOLLOW function, no more hands up. Am I doing something wrong?

Reply:
Okay, by using "handsup_base" instead of "handsup_enter", and changing the flag to 49 it's working.